decentraland/unity-explorer
# feat: scene emotes & animator support to sdk7

## What does this PR change?

Fixes #1078
Fixes #1237 
Fixes #1264 
Fixes #1104 
Fixes #1262 

In order to solve the problem, all the animations of the scenes need to be converted to `Animator` leaving behind the legacy `Animation` system. This is directly related to: https://github.com/decentraland/asset-bundle-converter/pull/94
The animations that are meant to be played in the avatar (scene emotes) cannot be differentiated from any other animation in the scene in the conversion process, so we have to convert everything, since legacy animations cannot be played on the player's avatar.
A new system has been implemented to support `Animator` for the `SDKAnimatorComponent`, but we will keep supporting legacy animations in case there is any.
Also, a new system has been implemented to download the asset bundles of the scene emote required by [triggerSceneEmote](https://js-sdk-toolchain.pages.dev/funcs/js_runtime_apis.__system_RestrictedActions_.triggerSceneEmote), so it can be later played by the existing systems.

## How to test the changes?



1. Clean asset bundle cache
2. Follow issue steps
3. Go to goerli plaza and go through all scenes that has animations. For example: (76,-3), (74,-9), (72,-8)
4. Check that animations are working as expected
5. Check that the player emotes are working as expected

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
